### PR TITLE
Suggested changes on top of #743

### PR DESCRIPTION
**Pull request by paulgb**

### CODE REVIEW

File: plane/src/typed_unix_socket/mod.rs
Comment (paulgb) [inline plane/src/typed_unix_socket/mod.rs:13]: The semantics here are: if ID is given, this is a request/response; otherwise this is an event.

This replaces both `IDedMessage` and `Wrapped{Client,Server}MessageType`.

File: plane/src/typed_unix_socket/mod.rs
Comment (paulgb) [inline plane/src/typed_unix_socket/mod.rs:36]: `random_string()` over `random_token()` is important here because a socket path can only be max 108 bytes ([ref](https://man7.org/linux/man-pages/man7/unix.7.html)) and it is borderline with `random_token()` which is longer.

Fortunately, there is absolutely zero chance of collision here because paths are already scoped by test.

**Review by paulgb — COMMENTED** [review]
Add some inline comments.

File: plane/src/typed_unix_socket/client.rs
Comment (paulgb) [inline plane/src/typed_unix_socket/client.rs:15]: I find `{Client,Server}MessageType` ambiguous because it's not clear whether a client message is a message sent by or to the client. I like the convention `MessageToX` or `MessageFromX` to make the distinction obvious.

File: plane/src/typed_unix_socket/server.rs
Comment (michaelsilver) [inline plane/src/typed_unix_socket/server.rs:84]: One thing I was intentional about with the API design is I'd like the details of the ID to be as hidden as possible from the calling code. As such, it'd be nice if `send_response` could accept the request and response like before, and internally take care of extracting/reusing the ID.

File: plane/src/typed_unix_socket/mod.rs
Comment (michaelsilver) [reply plane/src/typed_unix_socket/mod.rs:13]: > The semantics here are: if ID is given, this is a request/response; otherwise this is an event.

Nit: can we add a comment about this?

File: plane/src/typed_unix_socket/mod.rs
Comment (michaelsilver) [reply plane/src/typed_unix_socket/mod.rs:36]: Nit: seems like a good idea to put something like this in a comment.

File: plane/src/typed_unix_socket/mod.rs
Comment (michaelsilver) [inline plane/src/typed_unix_socket/mod.rs:56]: (This function call will be cleaner once we change `send_response` to simply take `(request, response)`)

File: plane/src/lib.rs
Comment (michaelsilver) [inline plane/src/lib.rs:27]: Good call

File: plane/src/typed_unix_socket/mod.rs
Comment (michaelsilver) [inline plane/src/typed_unix_socket/mod.rs:11]: Minor nit: I'm borderline on this one, but thoughts on creating a type (even if it's just a `String` alias) for the ID?

Comment (michaelsilver) [reply plane/src/typed_unix_socket/mod.rs:11]: It's less important once we change the signature of `send_response` but that was largely what motivated this suggestion -- something like `RandomID` is more meaningful of a type than `String`.

**Review by michaelsilver — APPROVED** [review]
Nice, appreciate it! Left a few small comments -- up to you if you want to address and merge or merge and leave it to me to tweak. Either way I want to unblock your larger review of the library so feel free to merge without requesting another review.

File: plane/src/typed_unix_socket/mod.rs
Comment (paulgb) [reply plane/src/typed_unix_socket/mod.rs:69]: My general preference is for functions to take just the data they need to do what they need to do. I find that this makes them easier to test and more versatile when refactoring, and the actual data flow is more accurately represented by the function signatures.

I'm inclined to keep this to ID, but use a newtype as you suggested to take a `MessageId` or similar as the first param instead of a `String`.

Comment (paulgb) [reply plane/src/typed_unix_socket/mod.rs:69]: Just I read your rationale below and I see that side of it too. I do like the idea of hiding the implementation detail of the ID, so I'm fine with that. It does mean that if we want to use the contents of the original message we will need to clone it though, since we won't be able to pass the request to `send_response` if we have already taken the message field from it.

**Review by michaelsilver — APPROVED** [review]

File: plane/src/typed_unix_socket/mod.rs
Comment (michaelsilver) [reply plane/src/typed_unix_socket/mod.rs:69]: Sounds good, I'll pick one of these options and finalize it in the parent PR!

**Pull request merged**
